### DISCUSSION TITLE
Диагностика LatinAndCyrillicSymbolInWordDiagnostic

### DISCUSSION THREAD

**Discussion by Denis-Miheykin** [1c-syntax/bsl-language-server#3295] [Ideas]
Добрый день.

Подскажите, можно ли в диагностике [LatinAndCyrillicSymbolInWordDiagnostic](https://github.com/1c-syntax/bsl-language-server/blob/develop/src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/LatinAndCyrillicSymbolInWordDiagnostic.java#L63) выполнить доработку таким образом, чтобы в параметр `excludeWords` можно было передать не конкретное слово, а некоторый шаблон? Например: `*XDTO`, `XDTO*` или `*XDTO*`. Такое пригодилось бы для общего модуля `МенеджерОбменаЧерезУниверсальныйФормат`.

**Comment by nixel2007** [comment]
Без отладки сложно сказать, но кажется, что из коробки такого нет, так как там используется квотирование входящего слова и есть символ поиска начала строки.

Но есть вот такой параметр:   allowTrailingPartsInAnotherLanguage, правда он глобальный.

А можно ли доработать - конечно можно.

**Reply by Denis-Miheykin** [reply]
> Но есть вот такой параметр: allowTrailingPartsInAnotherLanguage

Да, пробовал этот параметр и ожидал, что будет игнорировать процедуры с окончанием `XDTO`, но диагностика все равно срабатывает.

> А можно ли доработать - конечно можно.

👍

**Reply by nixel2007** [reply]
А вот это странно. Можно воспроизводимый пример? И как добавляли?

**Reply by Denis-Miheykin** [reply]
Настройка:
![image](https://github.com/1c-syntax/bsl-language-server/assets/20030975/370253d6-1615-4ca3-b3de-f1032dc2f08d)

Модуль:
![image](https://github.com/1c-syntax/bsl-language-server/assets/20030975/11afb503-4e6a-474e-80a6-70667dc56fc8)

**Reply by Denis-Miheykin** [reply]
Хотя вот странно, создал для теста пустой модуль и добавил 2 процедуры, диагностика не "ругается":
![image](https://github.com/1c-syntax/bsl-language-server/assets/20030975/a63c35f7-df85-42e1-aaac-3fb35cb51eab)

Диагностика лишь с флагом, убрал из первого параметра `XDTO`:
![image](https://github.com/1c-syntax/bsl-language-server/assets/20030975/ca176f5d-c97e-4ba4-96d8-226fad02cd84)

А рабочий модуль продолжает подсвечивать:
![image](https://github.com/1c-syntax/bsl-language-server/assets/20030975/3574eb0a-d187-4b60-95c6-7eb6add5751c)

**Reply by Denis-Miheykin** [reply]
Как я понял, ругается, если имя процедуры начинается со строчной буквы:
![image](https://github.com/1c-syntax/bsl-language-server/assets/20030975/21105435-e1ca-4fc8-9e86-1cc77df7fac6)

**Reply by nixel2007** [reply]
о как интересно... кажется, это надо конвертировать в ишуз.

**Reply by Denis-Miheykin** [reply]
Вот еще, если паттерн:
![image](https://github.com/1c-syntax/bsl-language-server/assets/20030975/15d552d5-9f22-4723-8512-a7a4972a6dee)
смотреть через браузер:
![image](https://github.com/1c-syntax/bsl-language-server/assets/20030975/7040223c-6a4f-4374-a235-e47811d0c6c7)
то имя процедуры под шаблон попадает, и по логике оно должно его игнорировать:
![image](https://github.com/1c-syntax/bsl-language-server/assets/20030975/06430c93-8860-4698-a418-239689160502)

Но вот интересно, должно ли имя процедуры попадать под паттерн? Ведь она должна начинаться с прописной буквы на en или ru:
![image](https://github.com/1c-syntax/bsl-language-server/assets/20030975/63c18590-c5f0-4166-b4d8-4a3d2faa674e)

Либо я ошибаюсь в паттерне.

**Reply by Denis-Miheykin** [reply]
> это надо конвертировать в ишуз.

Если есть причина, я не против :)

**Reply by nixel2007** [reply]
> Либо я ошибаюсь в паттерне.

все верно. осталось понять, зачем было сделано именно так =/

ишуз я уже создал. спасибо за расследование!
